### ISSUE TITLE
Dataset Preparation of Bridge_v1 and v2

### ISSUE THREAD

**Issue by taeyeopl** [Kiteretsu77/This_and_That_VDM#2]
Do I understand that your method starts from bridge v1, and v2 raw dataset for curation?

For bridge_data_v1, Does your method only use the Berkeley except upenn from here (https://rail.eecs.berkeley.edu/datasets/bridge_release/raw/bridge_data_v1/)? Is there any reason not to use upenn? 

For bridge_data_v2, use from here (https://rail.eecs.berkeley.edu/datasets/bridge_release/raw/bridge_data_v2/)

Do you have any plans to released the preprocessed training dataset?

**Comment by Kiteretsu77** [comment]
Yes, the v1 didn't use the Upenn, only Berkeley. It's just feeling that UPenn subdataset image quality is quite different from Berkeley, and the Berkeley samples are already a lot. The Bridge V2 is that one. I have released the preprocessed code. I don't plan to release the preprocessed training dataset because that is too big and I am not sure if the license of the Bridge allow us to do that. Thanks!

**Comment by taeyeopl** [comment]
Thanks @Kiteretsu77! Do you have any suggestions for downloading the dataset or using Python scripts? Manually clicking download for each image seems inefficient - is there a better solution?

**Comment by Kiteretsu77** [comment]
I think that using this link should download the full zip and then unzip will get everything: https://rail.eecs.berkeley.edu/datasets/bridge_release/data/

**Comment by taeyeopl** [comment]
Thanks @Kiteretsu77. Is it right to download the data zip files below the two datasets that get the same as your dataset instead of raw files (https://rail.eecs.berkeley.edu/datasets/bridge_release/raw/)? Do the below links consist of both Bridge v1 and v2? 

1. https://rail.eecs.berkeley.edu/datasets/bridge_release/data/demos_8_17.zip
2. https://rail.eecs.berkeley.edu/datasets/bridge_release/data/scripted_6_18.zip

**Comment by Kiteretsu77** [comment]
I remember that the zip file I downloaded starts with demos_8_17.zip, and then I unzip it and then choose the Bridge dataset inside with the V1 and V2 needed

**Comment by taeyeopl** [comment]
Thanks for the detailed explanation. I successfully download the dataset with your help. I realized that `demos_8_17.zip `contains `raw` files and `scripted_6_18.zip` contains `scripted_raw (which contains depth)`. 

Is there a specific reason for not using the` scripted raw `files?

Also, How long does it take to train the VL and VGL models provided in your  [training script](https://github.com/Kiteretsu77/This_and_That_VDM?tab=readme-ov-file#-training)? (It looks like about 44 hours for VL in my case.)

**Comment by Kiteretsu77** [comment]
No specific reason. I just choose one of them for the training. For me, When I used 8x Nvidia L40s, I trained like 2-3 days for the VL, and VGL is slightly shorter (because less iterations).

**Comment by taeyeopl** [comment]
Thanks for your replies. It was really helpful.  I'm curious about an alternative approach to obtaining the end effector center points/pose. Instead of using YOLOv8 for 2D center point detection, would it be possible to extract this information directly from the bridge dataset information (e.g.  `policy_out.pkl`, `obs_dict.pkl files`)? If possible, this method could potentially provide more accurate labels. It looks like they provide trajectories, but I'm unsure whether we could get the pose of the end effectors. What do you think?

**Comment by Kiteretsu77** [comment]
When we are doing the research on the pkl file, we tried to use existing information as much as possible to get the gripper location. We did use policy_out.pkl to get some start end frame index, but that is not spatial information that directly tells where in the pixel we can find the gripper, which leads us to train a YOLOv8. At that time, there isn't any SAM2, but I would recommend you to adjust the tracking by SAM2 instead of my hard-code tracking code, which is too hand-crafted comapared to SAM2 performence.

**Comment by taeyeopl** [comment]
Thanks for suggesting using SAM2. I also checked the `obs_dict.pkl` files. It has `eef_transform`, which looks like [end effector transform](https://github.com/rail-berkeley/bridge_data_robot/blob/b841131ecd512bafb303075bd8f8b677e0bf9f1f/widowx_envs/widowx_envs/base/robot_base_env.py#L306). Have you also tired to use information?

**Comment by taeyeopl** [comment]
Also, should I run the `curation_pipeline/tracking_by_keypoint.py` to train the vgl model? I think this is missing in the repository.

**Comment by Kiteretsu77** [comment]
I think that that file is deprecated. Based on readyme, you should run the curation_pipeline/select_frame_with_this_that.py for the tracking points.

**Comment by taeyeopl** [comment]
Thanks, it works for me. I have a question related to iteration. When I check the 30k iteration checkpoints of my VGL model, the results are weird. However, the 15k iteration checkpoint results look fine. What do you think? Have you encountered a similar problem?

[30k iteration results of VGL]
![combined](https://github.com/user-attachments/assets/59371ad1-fab6-488c-865e-ce7aab69238f)


[15k iteration results of VGL]
![combined](https://github.com/user-attachments/assets/bdc6e981-4dba-418b-a5c3-fb590dc93b5b)

**Comment by Kiteretsu77** [comment]
Is this your training result? If it is, I remember that I have also observed similar issue of very distorting videos when training much longer. However, it was like more than 50K. Did you strictly follow the training stage of first unet, then gesturenet? If it is, I would think that it is different setting variance. Thanks!

**Comment by taeyeopl** [comment]
Thanks for the comments. Right, this is my training results. I followed the provided scripts except `num_processes:4`, I changed to `num_processes:8` for training unet and gesturenet. 

https://github.com/Kiteretsu77/This_and_That_VDM/blob/b7ea2fb96c218a63c28b4504971828880c95b874/config/accelerate_config.json#L11
